axum-extra/src/multipart_builder.rs

```diff
@@ -0,0 +1,330 @@
+//! Generate forms to use in responses. You're probably looking for `MultipartForm`.
+
+use axum::response::{IntoResponse, Response};
+use fastrand;
+use http::{header, HeaderMap, StatusCode};
+use mime::Mime;
+
+/// The `Content-Transfer-Encoding` setting for a part.
+#[derive(Debug)]
+pub enum TransferEncoding {
+    /// If not specified, encoding defaults to UTF-8
+    TextUTF8,
+    /// If transferring raw binary data that is not guaranteed to be valid UTF-8.
+    Binary,
+}
+
+/// Create multipart forms to be used in API responses.
+/// This struct implements [IntoResponse], and so it can be returned from a handler.
+#[derive(Debug)]
+pub struct MultipartForm {
+    parts: Vec<Part>,
+}
+
+impl MultipartForm {
+    /// Construct a new empty multipart form with no parts.
+    pub fn new() -> Self {
+        MultipartForm { parts: Vec::new() }
+    }
```

Comment:
What use is this if there's no method to add a part?

Reply:
The ability to add a part was removed in a previous iteration, so this is stale, good catch :)